### DISCUSSION TITLE
Versioning of Collection Layouts?

### DISCUSSION THREAD

**Discussion by tosteiner** [knowledgefutures/pubpub#1831] [Support] [answered]
Hi all at PubPub, we at COPIM are using PubPub's `Collection > Book` template to publish multi-chapter report books, examples include e.g. the [Books Contains Multitudes](https://copim.pubpub.org/books-contain-multitudes-exploring-experimental-publishing) and [Interaction and Re-Use for OA Books](https://copim.pubpub.org/promoting-and-nurturing-interactions-with-open-access-books-strategies-for-publishers-and-authors) reports.

For each of these books, the editorial part has been realized via a Collection Layout, which consist of an overarching Editorial Framing of sorts (a Text Block), and is then followed by a block containing the chapter Pubs tagged with that collection. (PubPub team, see [Layout settings](https://copim.pubpub.org/dash/collection/promoting-and-nurturing-interactions-with-open-access-books-strategies-for-publishers-and-authors/layout))

 Now we are seeking to create a new revised & updated version of one of these reports and are wondering how to go about that in practical terms... like, is it possible to create a new version of the elements that are contained in a Collection's Layout (as e.g. the Text block would be changed for the revised version)? Or would we need to start with a whole new Collection Layout? (in short: are Collection Layouts versionable?)
 
 Apologies if this doesn't make sense, don't hesitate to let me know if you'd like me to describe this in a different way :)

**Comment by gabestein** [comment, marked as answer]
Makes total sense. Unfortunately, collection layouts are not versionable, so the best way to do what you're describing would be to create a new version of the layout. We're aware of this limitation, and have some initial ideas about fixing it involving making layouts and pages function similarly to pubs, but none of that work is yet specified, much less prioritized. More likely in the shorter term would be the ability to copy a page layout so you're not starting entirely from scratch, but that's also still being specified, I'm afraid.

One thing we could do to make it easier is manually copy the existing layout into a new one for you in the database. If that's of interest, feel free to write into help@ and we can get it taken care of pretty easily.

**Reply by tosteiner** [reply]
Thanks so much, Gabe, much appreciated! Just thought I'd check to see if there's something in the development pipeline that would help with this, but no worries at all, will discuss with the team, and am confident we can work with the manual copy route.
